File: src/main/java/net/shotbow/interestingfish/objects/Descriptor.java
package net.shotbow.interestingfish.objects;

import java.io.Serializable;

/**
 * User: Austin
 * Date: 5/18/13
 * Time: 1:06 AM
 * (c) lazertester
 */
public class Descriptor implements Serializable {

    private static final long serialVersionUID = 1;

    private String text;
    private double minWeightModifier;
    private double maxWeightModifier;
    private int rollWeight;

    public String getText() {
        return text;
    }

    public Descriptor setText(String text) {
        this.text = text;
        return this;
    }

    public double getMinWeightModifier() {
        return minWeightModifier;
    }

    public Descriptor setMinWeightModifier(double minWeightModifier) {
        this.minWeightModifier = minWeightModifier;
        return this;
    }

    public double getMaxWeightModifier() {
        return maxWeightModifier;
    }

    public Descriptor setMaxWeightModifier(double maxWeightModifier) {
        this.maxWeightModifier = maxWeightModifier;
        return this;
    }

    public int getRollWeight() {
        return rollWeight;
    }

    public Descriptor setRollWeight(int rollWeight) {
        this.rollWeight = rollWeight;
        return this;
    }
}

File: src/main/kotlin/net/shotbow/interestingfish/models/Descriptor.kt
package net.shotbow.interestingfish.models

import java.io.Serializable
import kotlin.random.Random

data class Descriptor(
    val text: String,
    val minWeightModifier: Double,
    val maxWeightModifier: Double,
    val rollWeight: Int
) : Serializable {

    val weight: Double
        get() = minWeightModifier + Random.nextDouble() * maxWeightModifier - minWeightModifier

    companion object {
        private const val serialVersionUID: Long = 1
    }
}
